src/polymer/declaration-property-handlers.ts

```diff
@@ -16,15 +16,48 @@ import * as estree from 'estree';
 
 import * as astValue from '../javascript/ast-value';
 import {JavaScriptDocument} from '../javascript/javascript-document';
-import {Severity} from '../warning/warning';
+import {ScannedBehaviorAssignment} from '../polymer/behavior';
+import {Severity, Warning} from '../warning/warning';
 
 import {analyzeProperties} from './analyze-properties';
 import {ScannedPolymerElement} from './polymer-element';
 
+export type BehaviorAssignmentOrWarning = {
+  kind: 'warning',
+  warning: Warning
+} | {kind: 'behaviorAssignment', assignment: ScannedBehaviorAssignment};
+
+export function getBehaviorAssignmentOrWarning(
```

Comment:
I prefer this style when I want to be sure that the caller is deliberate about both cases. Exceptions are tricky because they're so invisible, usually undocumented, and generally easy to forget to handle.